packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -168,6 +168,13 @@ export interface ClusterOptions {
    */
   readonly version?: string;
 
+  /**
+   * Enable or disable exporting the Kubernetes control plane logs for your cluster to CloudWatch Logs.
+   *
+   * @default - False, meaning that cluster control plane logs are not exported to CloudWatch Logs.
+   */
+  readonly controlPlaneLogging?: boolean | ControlPlaneLogging;
```

Comment:
Union types cannot be translated to certain languages and therefore not allowed in the CDK APIs.

Here's the pattern we have for this use case:

```ts
readonly controlPlaneLogging?: boolean;
readonly controlPlaneLoggingOptions?: ControlPlaneLogging;
```

- if `controlPlaneLogging` is `true` and `options` is undefined, then it will enable all logs.
- if `controlPlaneLoggingOptions` has a value (some config) and `controlPlaneLogging` is undefined, the it enables control plane logging with the specific config.
- if `controlPlaneLoggingOptions` has a value and `controlPlaneLogging` is explicitly `false` - raise an error.

This patterns offers an ergonomic experience:

```ts
// enable all logs (common use case)
new Cluster(this, 'cluster', { controlPlaneLogging: true });

// enable certain logs
new Cluster(this, 'cluster', {
  controlPlaneLoggingOptions: { ... }
});
```

Reply:
ok, so i will make the changes as suggested

Reply:
done. btw, thanks for the explanation!